src/pages/Auth/AuthPage.tsx

```diff
@@ -0,0 +1,171 @@
+  const showMsg = useAlert();
+  const [isLoading, setIsLoading] = useState(false);
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<IFormData>({
+    mode: 'onSubmit',
+    reValidateMode: 'onSubmit',
+  });
+
+  const onSubmit: SubmitHandler<IFormData> = async (data: IFormData) => {
+    if (isLogin) {
+      try {
+        setIsLoading(true);
+        await signInWithEmailAndPassword(authState, data.email, data.password);
+        showMsg({ type: 'success', content: localization.auth.loginSuccess });
+        navigate('/main');
+      } catch (e) {
+        if (e instanceof FirebaseError)
+          return showMsg({ type: 'error', content: fBaseErrReader(e.message) });
+      } finally {
+        setIsLoading(false);
+      }
+    } else {
+      try {
+        setIsLoading(true);
+        await createUserWithEmailAndPassword(authState, data.email, data.password);
+        showMsg({ type: 'success', content: localization.auth.accountCreated });
+      } catch (e) {
+        if (e instanceof FirebaseError)
+          return showMsg({ type: 'error', content: fBaseErrReader(e.message) });
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    reset();
+  };
+
+  return (
+    <Grid container direction="column" alignItems="center" justifyContent="center">
+      <CssBaseline />
+      <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square borderRadius={2}>
+        <PaperCss>
+          <Typography component="h1" variant="h5">
+            {isLogin ? localization.auth.login : localization.auth.signup}
+          </Typography>
+          <Form onSubmit={handleSubmit(onSubmit)}>
+            <InputField
+              variant="outlined"
+              margin="normal"
+              required
+              fullWidth
+              id="email"
+              label="Email"
+              autoComplete="email"
+              autoFocus
+              error={!!errors.email}
+              helperText={errors.email ? errors.email.message : ''}
+              {...register('email', {
+                required: true,
+                pattern: {
+                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
+                  message: 'invalid email address',
+                },
+              })}
+              InputLabelProps={{ shrink: true }}
+            />
+            <InputField
+              variant="outlined"
+              margin="normal"
+              fullWidth
+              label={localization.auth.password}
+              type="password"
+              id="password"
+              error={!!errors.password}
+              helperText={errors.password ? errors.password.message : ''}
+              {...register('password', {
+                required: true,
+                pattern: {
+                  value: /^(?=.*[0-9])(?=.*[!@#$%^&*])(?=.*[a-zA-Z])[a-zA-Z0-9!@#$%^&*]{8,}$/,
```

Comment:
https://www.computerhope.com/jargon/s/specchar.htm#:~:text=A%20special%20character%20is%20a,are%20examples%20of%20special%20characters. - it should be wider range of special characters.